extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java

```diff
@@ -1043,13 +1053,86 @@ void initialize(BuildProducer<SyntheticBeanBuildItem> syntheticBeans, QuteRecord
                         .map(GeneratedValueResolverBuildItem::getClassName).collect(Collectors.toList()), templates,
                         tags, variants))
                 .done());
-        ;
     }
 
     @BuildStep
     @Record(value = STATIC_INIT, optional = true)
     DevConsoleRouteBuildItem invokeEndpoint(QuteDevConsoleRecorder recorder) {
-        return new DevConsoleRouteBuildItem("preview", "POST", recorder.invokeHandler());
+        recorder.setupRenderer();
+        return new DevConsoleRouteBuildItem("preview", "POST", new Handler<RoutingContext>() {
+            @Override
+            public void handle(RoutingContext context) {
+                context.request().setExpectMultipart(true);
+                context.request().endHandler(new Handler<Void>() {
+                    @Override
+                    public void handle(Void ignore) {
+                        MultiMap form = context.request().formAttributes();
+                        String templatePath = form.get("template-path");
+                        String testJsonData = form.get("template-data");
+                        String contentType = null;
+                        String fileName = templatePath;
+                        int slashIdx = fileName.lastIndexOf('/');
+                        if (slashIdx != -1) {
+                            fileName = fileName.substring(slashIdx, fileName.length());
+                        }
+                        int dotIdx = fileName.lastIndexOf('.');
+                        if (dotIdx != -1) {
+                            String suffix = fileName.substring(dotIdx + 1, fileName.length());
+                            if (suffix.equalsIgnoreCase("json")) {
+                                contentType = Variant.APPLICATION_JSON;
+                            } else {
+                                contentType = URLConnection.getFileNameMap().getContentTypeFor(fileName);
+                            }
+                        }
+                        try {
+                            BiFunction<String, Object, String> renderer = DevConsoleManager
+                                    .getGlobal(QuteDevConsoleRecorder.RENDER_HANDLER);
+                            Object testData = Json.decodeValue(testJsonData);
```

Comment:
@stuartwdouglas So when and in which context is this method invoked? I'm asking because it needs `jackson-databind` on the class path...

Reply:
It is on the -deployment class path, which has a dependency on jackson-databind as it is not excluded on the deployment side.

Reply:
Hm, it's added on the -deployment class path because of `io.vertx:vertx-web-client:jar:test` > `io.vertx:vertx-core:jar:3.9.5:compile` > `com.fasterxml.jackson.core:jackson-databind:jar:2.11.3:compile` which is not very obvious... but ok (although I can't I understand all the class loading logic that is used here ;-).

Reply:
I will change it to be explicit.